.github/workflows/auto-updates.yaml

```diff
@@ -0,0 +1,65 @@
+name: Update translations and Rust packaging related files in main
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - debian/control
+concurrency: auto-update
+
+jobs:
+  update-rust-packaging:
+    name: Update packaging related Rust files
+    needs: update-po
+    runs-on: ubuntu-latest
+    # Right now, ubuntu 22.04 does not have the dh-cargo-vendored-sources script that is needed to
+    # run this job, so we need to run it inside a rolling container to get the latest version possible.
+    # This should be updated as soon as the dh-cargo version with the mentioned script gets ported to
+    # 22.04 or ubuntu-latest changes to a more recent version.
+    container:
+      image: ubuntu:rolling
+    steps:
+      - name: Install dependencies
+        run: |
+          DEBIAN_FRONTEND=noninteractive apt update
+          DEBIAN_FRONTEND=noninteractive apt install -y cargo dh-cargo git jq
+      - uses: actions/checkout@v4
+        with:
+          ref: main
+      - name: Vendor the dependencies
+        run: |
+          cargo vendor vendor_rust/
+      - name: Update XS-Vendored-Sources-Rust
+        run: |
+          set -eu
+
+          export CARGO_VENDOR_DIR=vendor_rust/
+          VENDORED_SOURCES=$(/usr/share/cargo/bin/dh-cargo-vendored-sources 2>&1 || true)
+          OUTPUT=$(echo "$VENDORED_SOURCES" | grep ^XS-Vendored-Sources-Rust: || true)
+          if [ -z "$OUTPUT" ]; then
+            echo "XS-Vendored-Sources-Rust is up to date. No change is needed.";
+            exit 0
+          fi
+          sed -i "s/^XS-Vendored-Sources-Rust:.*/$OUTPUT/" debian/control
```

Comment:
Well the multiline output could be generated with:

       echo "$OUTPUT" | grep-dctrl -s XS-Vendored-Sources-Rust -n - | sed "s|,\([^,]*\)|\n                          \1|g"

Now w just need to inject it properly :dancers:

Reply:
It will support it, but it's fine, it's not a big deal since these are auto-generated and we'll trust that CI will help us maintaining it properly :)

---

```perl
my $actual_raw = $source->get_custom_field('Vendored-Sources-Rust');
my @actual = sort(split(/\s*,\s*/, $actual_raw // ""));
```